viewer/packages/cad-geometry-loaders/src/sector/culling/ByScreenSizeSectorCuller.ts

```diff
@@ -0,0 +1,215 @@
+): { model: CadModelMetadata; sectorId: number; priority: number }[] {
+  const candidateSectors = new Array<{
+    model: CadModelMetadata;
+    sectorId: number;
+    priority: number;
+  }>();
+  for (const [model, sectors] of modelsAndCandidateSectors) {
+    sectors.forEach(sectorMetadata => {
+      const sector = sectorMetadata;
+      weightFunctions.computeTransformedSectorBounds(sector.bounds, model.modelMatrix, transformedBounds);
+      const priority = determineSectorPriority(weightFunctions, sector, transformedBounds);
+      candidateSectors.push({
+        model,
+        sectorId: sector.id,
+        priority
+      });
+    });
+  }
+  candidateSectors.sort((left, right) => {
+    return right.priority - left.priority;
+  });
+  return candidateSectors;
+}
+
+/**
+ * Determines candidate sectors per model, i.e. sectors within frustum.
+ */
+function determineCandidateSectorsByModel(
+  cadModelsMetadata: CadModelMetadata[],
+  cameraWorldInverseMatrix: THREE.Matrix4,
+  cameraProjectionMatrix: THREE.Matrix4,
+  input: DetermineSectorsInput
+) {
+  return cadModelsMetadata.reduce((result, model) => {
+    const sectors = determineCandidateSectors(
+      cameraWorldInverseMatrix,
+      cameraProjectionMatrix,
+      model.modelMatrix,
+      model.scene,
+      input.clippingPlanes
+    );
+    result.set(model, sectors);
+    return result;
+  }, new Map<CadModelMetadata, V9SectorMetadata[]>());
+}
+
+/**
+ * Prepares data structures with model and sectors
+ */
+function initializeTakenSectorsAndWeightFunctions(
+  modelsAndCandidateSectors: Map<CadModelMetadata, V9SectorMetadata[]>,
+  takenSectors: TakenV9SectorMap,
+  weightFunctions: WeightFunctionsHelper
+) {
+  for (const [model, sectors] of modelsAndCandidateSectors) {
+    takenSectors.initializeScene(model);
+    weightFunctions.addCandidateSectors(sectors, model.modelMatrix);
+  }
+}
+
+/**
+ * Determines candidate sectors, i.e.
+ */
+function determineCandidateSectors(
+  cameraWorldInverseMatrix: THREE.Matrix4,
+  cameraProjectionMatrix: THREE.Matrix4,
+  modelMatrix: THREE.Matrix4,
+  modelScene: SectorScene,
+  clippingPlanes: THREE.Plane[]
+): V9SectorMetadata[] {
+  if (modelScene.version !== 9) {
+    throw new Error(`Expected model version 9, but got ${modelScene.version}`);
+  }
+
+  const transformedCameraMatrixWorldInverse = new THREE.Matrix4();
+  transformedCameraMatrixWorldInverse.multiplyMatrices(cameraWorldInverseMatrix, modelMatrix);
+  const sectors = modelScene
+    .getSectorsIntersectingFrustum(cameraProjectionMatrix, transformedCameraMatrixWorldInverse)
+    .map(x => x as V9SectorMetadata);
+
+  if (clippingPlanes.length > 0) {
+    const bounds = new THREE.Box3();
+    return sectors.filter(sector => {
+      bounds.copy(sector.bounds);
+      bounds.applyMatrix4(modelMatrix);
+
+      const shouldKeep = clippingPlanes.every(plane => isBox3OnPositiveSideOfPlane(bounds, plane));
+      return shouldKeep;
+    });
+  }
+  return sectors;
```

Comment:
I invert the if in cases like this to reduce branching:
```suggestion
  if (clippingPlanes.length <= 0) {
    return sectors;
  }
  
  const bounds = new THREE.Box3();
 
  return sectors.filter(sector => {
    bounds.copy(sector.bounds);
    bounds.applyMatrix4(modelMatrix);

    const shouldKeep = clippingPlanes.every(plane => isBox3OnPositiveSideOfPlane(bounds, plane));
    return shouldKeep;
  });
  
  
```